src/overlays/actors/ovl_En_Nwc/z_en_nwc.c

```diff
@@ -21,25 +51,200 @@ const ActorInit En_Nwc_InitVars = {
     (ActorFunc)EnNwc_Update,
     (ActorFunc)EnNwc_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Nwc/func_80ABC0E0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Nwc/func_80ABC0E8.s")
+static ColliderJntSphItemInit sJntSphElementInit = {
+    { 0x01, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+    { 0, { { 0, 0, 0 }, 10 }, 100 },
+};
+
+static ColliderJntSphInit_Set3 sJntSphInit = {
+    { COLTYPE_UNK3, 0x00, 0x09, 0x39, COLSHAPE_JNTSPH },
+    16, NULL,
+};
+
+void EnNwc_SetUpdate(EnNwc *this, EnNwcUpdateFunc updateFunc) {
+    this->updateFunc = updateFunc;
+}
+
+void EnNwc_ChickNoop(EnNwcChick *chick, EnNwc *this, GlobalContext *globalCtx) {
+}
+
+void EnNwc_ChickBgCheck(EnNwcChick *chick, GlobalContext *globalCtx) {
+    CollisionPoly *outPoly;
+    s32 bgId;
+    Vec3f outPos;
+    f32 dy;
+
+    chick->bgFlags &= ~CHICK_BG_WALL & ~CHICK_BG_FLOOR;
+    outPos.x = chick->pos.x;
+    outPos.y = chick->pos.y;
+    outPos.z = chick->pos.z;
+    if (func_8003D464(&globalCtx->colCtx, &outPos, &chick->pos, &chick->lastPos, 10.0f, &chick->floorPoly, 20.0f)) {
+        chick->bgFlags |= CHICK_BG_WALL;
+    }
+    //! @bug The use of outPos here is totally wrong. Even if it didn't get overwritten
+    //       by the wall check, it should add an offset to the y-value so the raycast
+    //       doesn't go through the floor and cause the chicks to ignore all floors.
+    chick->floorY = func_8003C940(&globalCtx->colCtx, &outPoly, &bgId, &outPos);
+    dy = chick->floorY - chick->pos.y;
+    if ((0.0f <= dy) && (dy < 40.0f)) {
+        chick->pos.y = chick->floorY;
+        chick->bgFlags |= CHICK_BG_FLOOR;
+    }
+}
+
+void EnNwc_ChickFall(EnNwcChick *chick, EnNwc *this, GlobalContext *globalCtx) {
+    chick->velY -= 0.1f;
```

Comment:
formatter

Reply:
Formatted. Also formatted the rest of the repo accidentally.